ui/raidboss/data/00-misc/test.ts

```diff
@@ -325,10 +327,36 @@ const triggerSet: TriggerSet<Data> = {
         },
       },
     },
+    {
+      id: 'Test Combatant Cast Enable',
```

Comment:
Just reiterating https://github.com/quisquous/cactbot/pull/5379#issuecomment-1546997059, but could you add a unit test for these lines as well?

Reply:
Adding tests is giving me some hassle because of the reverse checking, where `regexUtil` checks the returned capture groups to ensure they're valid params or something along those lines.

```
  1) netregex tests
       combatantMemory:
     Error: CombatantMemory: invalid parameter 'pairBNpcID'.  Valid params: ["capture","type","timestamp","change","id","pair"]
      at Function.validateParams (file:///mnt/c/Users/valarnin/Source/cactbot/resources/regexes.ts:694:15)
      at parseHelper (file:///mnt/c/Users/valarnin/Source/cactbot/resources/netregexes.ts:156:11)
      at buildRegex (file:///mnt/c/Users/valarnin/Source/cactbot/resources/netregexes.ts:336:10)
      at Function.combatantMemory (file:///mnt/c/Users/valarnin/Source/cactbot/resources/netregexes.ts:597:12)
      at file:///mnt/c/Users/valarnin/Source/cactbot/test/unittests/netregex_test.ts:415:63
      at default (file:///mnt/c/Users/valarnin/Source/cactbot/test/helper/regex_util.ts:61:34)
      at Context.<anonymous> (file:///mnt/c/Users/valarnin/Source/cactbot/test/unittests/netregex_test.ts:415:5)
      at processImmediate (node:internal/timers:471:21)
```

I'm not sure if it's really feasible to re-jigger the repeating fields defs to not require the `pair` array, I guess I can investigate that tomorrow.

Reply:
Yeah, feel free to special case anything here that needs to be special-cased and leave a TODO or some comments.  Mostly I just wanted some examples that exercise this code path.